test/test_tomopy/test_misc/test_corr.py

```diff
@@ -66,22 +73,43 @@ def test_gaussian_filter(self):
     def test_median_filter(self):
         loop_dim(median_filter, read_file('cube.npy'))
 
+    def test_median_filter_nonfinite(self):
+        # Add some random non-finite values to an array of all ones
+        data_org = np.ones(shape=(100, 100, 100))
+        for i in range(50):
+            x = np.random.randint(0, 100)
+            y = np.random.randint(0, 100)
+            z = np.random.randint(0, 100)
+            data_org[z, x, y] = np.inf
+            x = np.random.randint(0, 100)
+            y = np.random.randint(0, 100)
+            z = np.random.randint(0, 100)
+            data_org[z, x, y] = -np.inf
+            x = np.random.randint(0, 100)
+            y = np.random.randint(0, 100)
+            z = np.random.randint(0, 100)
+            data_org[z, x, y] = np.nan
+
+        data_post_corr = median_filter_nonfinite(
+            data_org.copy(),
+            size=5,
+            callback=None,
+        )
+        # All the post filtering values should be 1 because all of the finite
+        # values are 1.
+        assert np.all(data_post_corr == 1.0)
```

Comment:
This test could be improved. It should test a small array with a known solution (check the other tests in this module for examples) instead of being a random test with an unknown solution.

Reply:
When coming up with a better test I realized this function breaks if the entire kernel contains non-finite values. I am wondering what you think of the following before I build in error handling.

1) checks if each kernel only contains non-finite values
2) if it does then throw a ValueError to the user asking them to increase kernel size

Reply:
Good thinking! Agreed. You can check the length of `kernel_cropped_data[np.isfinite(kernel_cropped_data)]` greater than zero then raise a ValueError. I don't think a warning would be appropriate because returning any non-finite values in the array would mean the function doesn't do as claimed.

Remember to add a `Raises` section to the docstring explaining when a ValueError is raised.